### DISCUSSION TITLE
$route undefined Error in SSR mode

### DISCUSSION THREAD

**Discussion by mecjos** [quasarframework/quasar#13039] [CLI - SSR mode]
I'm new to that ssr thing and I get errors that doesn't make sense.. I was working without any problem, shut down the terminal and run again then I get $route undefined error from following line... didn't change anything related with this in my code. I will go crazy. when I shift to spa mode it works.. what's wrong with ssr mode ??

`<div class="flex flex-center" v-if="!showText && this.$route.path !== nav.route" key="icon"><img :src="'images/feature/' + nav.img" height="60" /></div>`

full error text is 

> [Quasar Dev Webserver] /features/manufacturing -> error during render
> TypeError: Cannot read properties of undefined (reading '$route')

**Comment by rstoenescu** [comment]
Please post a reproduction repo or a https://codesandbox.quasar.dev with a minimal reproduction so we can help you get through with your problem.

**Reply by mecjos** [reply]
> Please post a reproduction repo or a https://codesandbox.quasar.dev with a minimal reproduction so we can help you get through with your problem.

can I run ssr mode in codesandbox ?

**Comment by pdanpdan** [comment]
You are using this in template

**Reply by mecjos** [reply]
yes but I was already using.. It was working.

**Comment by pdanpdan** [comment]
The error message is clear about trying to access $route from this which is undefined

**Reply by mecjos** [reply]
Ok, thank you. it's understood.. it's undefined at server side..

**Comment by Lalrinfela** [comment]
Hi,

I think i found the problem.

Even on a fresh quasar project using Vuex alone on a webpack, it seems we need to install pinia dependencies.

The reason is when i run `quasar new store app` it installed pinia and now the store is working.

Seems, **vuex require pinia**

<img width="716" alt="Screenshot 2022-04-03 at 7 58 21 AM" src="https://user-images.githubusercontent.com/1894559/161408592-956eea25-28c6-4c2e-8317-a62121b0bffd.png">

Thanks everyone.

**Reply by rstoenescu** [reply]
No, it means that you haven't actually selected Vuex or Pinia when you scaffolded your app. The CLI checks if you have Pinia or Vuex installed before adding your new store. If none is installed, then it defaults to Pinia.
Vuex does NOT requires Pinia.

**Reply by mecjos** [reply]
Sorry but I couldn't understand what is the relation with vuex..:) I don't use vuex store in my project.

**Reply by metalsadman** [reply]
@Lalrinfela posted in wrong discussion https://github.com/quasarframework/quasar/discussions/13035
